packages/graphql-mapping-template/src/dynamodb.ts

```diff
@@ -288,10 +284,12 @@ export class DynamoDBMappingTemplate {
     ]);
   }
 
-  public static dynamoDBResponse(
-    expression: Expression = ref('util.error($ctx.error.message, $ctx.error.type, $ctx.result)')
-  ): CompoundExpressionNode {
-    return compoundExpression([ifElse(ref('ctx.error'), expression, ref('util.toJson($ctx.result)'))]);
+  public static dynamoDBResponse(isSyncEnabled: boolean, returnExpression?: Expression): CompoundExpressionNode {
+    const errorExpresion = isSyncEnabled
+      ? ref('util.error($ctx.error.message, $ctx.error.type, $ctx.result)')
+      : ref('util.error($ctx.error.message, $ctx.error.type)');
+    const resultExpression = returnExpression ? returnExpression : ref('util.toJson($ctx.result)');
```

Comment:
Small nit: For readability it's best to keep formatting strategy with line breaks consistent in a codebase, so if one ternary expression puts the cases on separate lines, the subsequent should do the same.